fennel/datasets/datasets.py

```diff
@@ -473,6 +475,13 @@ def first(self) -> _Node:
             self.keys = self.keys[0]  # type: ignore
         return First(self.node, list(self.keys))  # type: ignore
 
+    def window(self, type: str, gap: str, field: str) -> _Node:
```

Comment:
the type is not str but the enum right ?

Reply:
I actually thought `str` might be better because enums don't feel pythonic. For instance, `how` parameter of join takes a string, not enum (and same in pandas too). 

In fact, I've asked @zsid60 to change one of the enums in Kinesis connector to use str instead of enums.

Reply:
I think initiation should be from str, we are internally converting str to Enum, but if method has Enum, then that's an extra import user has to do.